integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/NewCarbonDataLoadRDD.scala

```diff
@@ -0,0 +1,281 @@
+    val iter = new Iterator[(K, V)] {
+      var partitionID = "0"
+      val loadMetadataDetails = new LoadMetadataDetails()
+      var model: CarbonLoadModel = _
+      var uniqueLoadStatusId =
+        carbonLoadModel.getTableName + CarbonCommonConstants.UNDERSCORE + theSplit.index
+      try {
+        loadMetadataDetails.setPartitionCount(partitionID)
+        loadMetadataDetails.setLoadStatus(CarbonCommonConstants.STORE_LOADSTATUS_FAILURE)
+
+        carbonLoadModel.setSegmentId(String.valueOf(loadCount))
+        val recordReaders = getRecordReaders
+        val loader = new SparkPartitionLoader(model,
+          theSplit.index,
+          hdfsStoreLocation,
+          kettleHomePath,
+          loadCount,
+          loadMetadataDetails)
+        // Intialize to set carbon properties
+        loader.initialize()
+        loadMetadataDetails.setLoadStatus(CarbonCommonConstants.STORE_LOADSTATUS_SUCCESS)
+        CarbonLoaderUtil.executeNewDataLoad(model,
+          loader.storeLocation,
+          hdfsStoreLocation,
+          recordReaders)
+      } catch {
+        case e: Exception =>
+          logInfo("DataLoad failure")
+          LOGGER.error(e)
+          throw e
+      }
+
+      def getRecordReaders: Array[RecordReader[NullWritable, StringArrayWritable]] = {
+        val attemptId = newTaskAttemptID(jobTrackerId, id, isMap = true, theSplit.index, 0)
+        val configuration: Configuration = confBroadcast.value.value
+        configureCSVInputFormat(configuration)
+        val hadoopAttemptContext = newTaskAttemptContext(configuration, attemptId)
+        val format = new CSVInputFormat
+        if (isTableSplitPartition) {
+          // for table split partition
+          val split = theSplit.asInstanceOf[CarbonTableSplitPartition]
+          logInfo("Input split: " + split.serializableHadoopSplit.value)
+          carbonLoadModel.setTaskNo(String.valueOf(theSplit.index))
+          if (carbonLoadModel.isDirectLoad) {
+            model = carbonLoadModel.getCopyWithPartition(
+                split.serializableHadoopSplit.value.getPartition.getUniqueID,
+                split.serializableHadoopSplit.value.getPartition.getFilesPath,
+                carbonLoadModel.getCsvHeader, carbonLoadModel.getCsvDelimiter)
+          } else {
+            model = carbonLoadModel.getCopyWithPartition(
+                split.serializableHadoopSplit.value.getPartition.getUniqueID)
+          }
+          partitionID = split.serializableHadoopSplit.value.getPartition.getUniqueID
+
+          StandardLogService.setThreadName(partitionID, null)
+          CarbonTimeStatisticsFactory.getLoadStatisticsInstance.recordPartitionBlockMap(
+              partitionID, split.partitionBlocksDetail.length)
+          val readers =
+          split.partitionBlocksDetail.map(format.createRecordReader(_, hadoopAttemptContext))
+          readers.zipWithIndex
+            .foreach(f => f._1.initialize(split.partitionBlocksDetail(f._2), hadoopAttemptContext))
```

Comment:
change `f` to `case (reader, index)` to make it more readable
move foreach to previous line